datasets/NRS-Weekly-Data-on-Births-and-Deaths-Registered-in-Scotland/Spec.md

```diff
@@ -0,0 +1,45 @@
+
+## Actions to Take
+* Alter the `Week Number` - remove the preceding `W` so we're left with just a number between 1 and 53.
+* Remove all values for Week 53 where the cell value is `-`. These years do not have a 53rd week.
+* The `Area` should be mapped to the ONS geography code for Scotland `http://statistics.data.gov.uk/id/statistical-geography/S01011834`.
+    * This should be taken care of by my changes to the `info.json` file.
+* Rename the `OBS` column to `Value`.
+
+## Issues
+
+* Information is encoded in the XLSX file by *italicising* text. Unfortunately we can't extract this information using databaker as it stands. 
+
+
+
+
+## Leigh notes
+
+
+	Year column needs to be converted to an integer
+	Week and year column need to be combined into format week/{year}-{week} - week/2020-02
+	Column needs a name like Period
+	Value (OBS) column needs to be converted to integer
+	Measure Type and Unit column can be removed as they will be taken care of in the info.json (this will hopefully change soon with the use of the cube class)
+
+	If still can't do multi-measure cubes
```

Comment:
Is this truly a multi-measure cube given that each record only has one of births/deaths? I understood it that the problem we had with multi-measure cubes was that we hadn't figured the right ontological representation for them yet?

Reply:
The issue with multi-measures is that there are a few different ways to represent them in the Tidy CSV:
1. one row per observation, one column per dimension and one distinct value column per measure type.
2. one row per observation, a column for each dimension as above and then one column for each of "Measure Type", "Unit" and "Value". This is what we were doing for targeting PMD3.
3. one Tidy CSV file per measure, where each CSV file has one row per observation, one column per dimension as usual, then one value column with the measure/unit type for that file, e.g. one file would be observations-weight.csv, the other observations-volume.csv. These separate CSV files would then be amalgamated into the same data cube with multiple measures.

There are also a couple of different ways to represent multiple measures in the Data Cube, though we've chosen https://www.w3.org/TR/vocab-data-cube/#dfn-measure-dimension as the main way to do this, partly because it fits with (2) above.

As I understand this dataset, there are two measures for each fixed value of the dimensions (observation): births and deaths.

Reply:
I think we're looking at this from two different perspectives. 

I'm looking at it from the perspective of having one dataset with two rows:

Area | Year | Week | Unit
Scotland | 1974 | 1 | Births 
Scotland | 1974 | 1 | Deaths 

If we import this dataset in this way we would create two seperate observations, both of them with unit `Count`, but each one would have a different unit. One of them would be `Births` and the other `Deaths`. I don't see any schematic restriction on doing this.

Under your approach we would create two seperate datasets with two seperate info.json files and two seperate schemas but we'd end up with exactly the same number of observations.

I guess what I'm really asking here is whether my approach is possible or whether there's some other restriction I'm missing?

Reply:
I think we're conflating measure and unit. In these observations, the units are "people", while the measures are "births" and "deaths".

We could encode the above as:

Area | Year | Week | Measure Type | Unit | Value
-----|------|------|--------------|------|------
Scotland | 1974 | 1 | Births | People | 871
Scotland | 1974 | 1 | Deaths | People | 906

or:

Area | Year | Week | Births | Deaths
-----|------|------|-------|------
Scotland | 1974 | 1 | 871 | 906

or two data frames, one for births, one for deaths:

Area | Year | Week | Births
-----|------|------|-------
Scotland | 1974 | 1 | 871

Area | Year | Week | Deaths
-----|------|------|-------
Scotland | 1974 | 1 | 871

With the last two being the way that Mike's Cubes() approach will let us turn things into a single Data Cube with two measures.

Reply:
Okay, so  for instance this example isn't quite right?

```
            "Value": {
                "unit": "http://gss-data.org.uk/def/concept/measurement-units/deaths",
                "measure": "http://gss-data.org.uk/def/measure/count",
                "datatype": "integer"
            }
```

https://github.com/GSS-Cogs/family-covid-19/blob/master/datasets/NHS-E-COVID-19-Daily-Deaths/info.json#L36